### ISSUE TITLE
Inotify leak

### ISSUE THREAD

**Issue by aj3423** [wez/wezterm#4498]
### What Operating System(s) are you seeing this problem on?

Linux Wayland

### Which Wayland compositor or X11 Window manager(s) are you using?

Fedora 39 KDE Wayland
kernel 6.5.6-300.fc39.x86_64

### WezTerm version

20231023_015614_84ad0400

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

I always keep 1 instance of wezterm running a server in the background, and I repeatedly open and close new wezterm window(one off). After a period of time the system shows notification like:
> inotify instance capacity low

I tried `kde-inotify-survey`, it shows:
```
    {
            "cmdline": "/usr/bin/wezterm-gui",
            "instances": 134,    <--------------------- leak
            "pid": 208637,
            "uid": 0,
            "watches": 45
    }
``` 
This "instances" increases by 1 every time a new wezterm window is launched, but it never decreases as long as there is one wezterm window left. But I cannot close them all because I need to keep the server running.

And I noticed that, when I close the last window, it consumes a lot CPU, one CPU core goes to 100% for several seconds before the window disappears, the more wezterm instances I've opened/closed previously, the longer it takes to exit. Maybe this is related to the inotify leak.

### To Reproduce

1. open new wezterm window
2. check the instances count of "wezterm-gui" with `kde-inotify-survey`
3. repeat step 1 and 2

Note: it can't be reproduced with .AppImage, it uses different "wezterm-gui" processes all the time.

### Configuration

```lua
return {
	default_workspace = "main",
}
```

### Expected Behavior

no leak

### Logs

Open two windows, and then close both.

[wezterm-gui-log-223866.txt](https://github.com/wez/wezterm/files/13194683/wezterm-gui-log-223866.txt)
[wezterm-gui-log-224022.txt](https://github.com/wez/wezterm/files/13194684/wezterm-gui-log-224022.txt)

**Comment by wez** [comment]
I can't reproduce this. I'm on Fedora 38.  Opening/closing windows, reloading the config doesn't have any impact on the number of instances I see:

```console
; kde-inotify-survey | grep -C 3 wezterm-gui
            "watches": 1
        },
        {
            "cmdline": "/usr/bin/wezterm-gui",
            "instances": 1,
            "pid": 11951,
            "uid": 1000,
```

Is there something else specific to your environment that is influencing this?
Is there something special about your home directory structure (which is what wezterm is trying to watch) that might influence this?
Please try to dig in to find out why this is happening for you!

**Comment by aj3423** [comment]
@wez Strange, I just tried F38 kde in VirtualBox, it doesn't have this problem, there isn't any entry of `/usr/bin/wezterm-gui` at all.  But on my F39, when there are 40 "instances":
> {
            "cmdline": "/usr/bin/wezterm-gui",
            "instances": 40,
            "pid": 391290,
            "uid": 0,
            "watches": 41
        }

I tried `ls -al /proc/391290/fd` to see what these 40 instances are, it looks like:

![image](https://github.com/wez/wezterm/assets/4710875/0e30850b-b518-4d3d-8ad8-1b6364dc2108)
There are 40 lines of `anon_inode:inotify` and 43 lines of `/dev/dri/renderD128`, I googled about `renderD128`, only found it's something about 3D rendering, any idea what is it?

**Comment by wez** [comment]
Maybe this is a graphics driver issue and not directly a wezterm issue? That would explain the correlation with multiple windows.
I'd suggest playing around with https://wezfurlong.org/wezterm/config/lua/config/front_end.html to see if switching between OpenGL, WebGgu (and with webgpu: try different gpus with https://wezfurlong.org/wezterm/config/lua/config/webgpu_preferred_adapter.html) to see if the issue goes away.

**Comment by aj3423** [comment]
@wez Seems it's caused by MangoHud. I just upgraded from kernel `6.5.6` to `6.5.9` and suddenly the window becomes:
![image](https://github.com/wez/wezterm/assets/4710875/dab18979-459f-4f53-9d0e-d9415af1c9a9)
I don't know why it didn't show the mangohud before the upgrade. Then I removed the `MANGOHUD=1` in my global environment variable, the problem is gone.
Thank you for the support and the awesome WezTerm.
